src/internal/connector/onedrive/collections.go

```diff
@@ -434,15 +466,62 @@ func (c *Collections) UpdateCollections(
 			// already created and partially populated.
 			updatePath(newPaths, *item.GetId(), folderPath.String())
 
+			if ok {
```

Comment:
We don't need this if check. We can just run the code below in all cases though maybe only make a new collection if it was moved so we don't create new empty collections accidentally

If we keep the if check I think there's an edge case where the user made a new folder (at some point, maybe between backups), we've picked up the new folder in the results but haven't gotten all results, the user moves the folder, and then we get a new result with the folder move. I'm not sure if Graph will ever return a sequence like that, but it may be best if we plan on handling it since it would be difficult to track down

Reply:
Looking at the graph API response, if we create a folder, create a file in it, then rename the folder to a separate name, we just get two delta entries one for the folder with the new name and one for the file with the path as the new location of the folder.